### DISCUSSION TITLE
correct syntax for DARE kms configuration in sftpgo.json?

### DISCUSSION THREAD

**Discussion by OliverAbraham** [drakkan/sftpgo#379] [Q&A] [answered]
Hello,
I'm trying to use the data at rest encryption (DARE) feature with a hard coded password.
But it has no effect. Files are still stored unencrypted.

I've created a file named "DareMasterKey" with a password. And my configuration is:
  "kms": {
    "secrets": {
      "url": "",
      "master_key_path": "/var/lib/sftpgo/DareMasterKey",
    }
  }

What am I doing wrong?

Best regards
Oliver

**Comment by drakkan** [comment, marked as answer]
Hi,

you are setting the kms master key, please take a look [here](https://github.com/drakkan/sftpgo/blob/main/docs/kms.md) for more details.

To use the `cryptfs` you have to enable it on a per-user basis, for example using the web admin, take a look at the following screenshot:

![Schermata del 2021-04-18 15-37-18](https://user-images.githubusercontent.com/553263/115147587-fce0ef80-a05b-11eb-8986-19499e344c69.png)

each file will be encrypted using a different [key](https://github.com/drakkan/sftpgo/blob/main/docs/dare.md)
